lahja/asyncio/endpoint.py

```diff
@@ -403,12 +422,18 @@ def run(self, *args, **kwargs):  # type: ignore
         # the Endpoint on the other end blocks until it receives this message
         await remote.notify_subscriptions_updated(self.subscribed_events)
 
+        await self._add_half_connection(remote)
+
     async def _handle_client(self, remote: RemoteEndpoint) -> None:
         try:
             async with run_remote_endpoint(remote):
-                await remote.wait_stopped()
+                try:
+                    await remote.wait_stopped()
+                except asyncio.CancelledError:
+                    return
```

Comment:
Why does this suppress `CancelledError`? I think that in this situation it's okay but letting it propagate up is safer.

From [the docs](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task):
> cancelled() can be used to check if the Task was cancelled. The method returns True if the wrapped coroutine did not suppress the CancelledError exception and was actually cancelled.

Suppressing the error prevents callers of `_handle_client` which didn't use `ensure_future` from knowing they've been cancelled, and it also prevents other callers of `_handle_client` from determining whether the coro stopped because of cancellation. Neither kind of caller exists right now but this still seems like unexpected behavior.